mountpoint-s3/src/main.rs

```diff
@@ -485,13 +490,30 @@ fn parse_perm_bits(perm_bit_str: &str) -> Result<u16, anyhow::Error> {
     }
 }
 
-pub fn retrieve_instance_type() -> anyhow::Result<String> {
+fn calculate_network_throughput() -> anyhow::Result<f64> {
+    let instance_type = retrieve_instance_type().context("failed to retrieve instance type")?;
+    let throughput = get_maximum_network_throughput(&instance_type).context("failed to get network throughput")?;
+    Ok(throughput)
+}
+
+fn retrieve_instance_type() -> anyhow::Result<String> {
     let imds_crt_client = ImdsCrtClient::new().context("failed to create IMDS client")?;
 
     let query = imds_crt_client
         .make_instance_type_query()
         .context("failed to send IMDS query")?;
 
     let result = futures::executor::block_on(query).context("IMDS query failed")?;
+    tracing::debug!("detected EC2 instance type {result}");
     Ok(result)
 }
+
+fn get_maximum_network_throughput(ec2_instance_type: &str) -> anyhow::Result<f64> {
+    let file = include_str!("../scripts/network_performance.json");
+    let throughput: serde_json::Value =
+        serde_json::from_str(file).context("failed to parse network_performance.json")?;
+    throughput
+        .get(ec2_instance_type)
+        .and_then(|t| t.as_f64())
+        .ok_or_else(|| anyhow!("no throughput configuration for EC2 instance type {ec2_instance_type}"))
+}
```

Comment:
Let's write a few unit tests for this function, since instance types never change over time. Probably check these ones:
* `c4.large` (`Moderate` in the API, so what should the result be?)
* `c5.large` (`Up to 10 Gigabit` in the API)
* `c5n.large` (`Up to 25 Gigabit`)
* `c6i.large` (`Up to 12.5 Gigabit`)
* `c5n.18xlarge` (`100 Gigabit`)
* `p4d.24xlarge` (`4x 100 Gigabit`, what should we do here?)

Reply:
we need to deal with `p4d.24xlarge` `dl1.24xlarge`  and `trn1.32xlarge` given the providing throughput literally like `4x 100 Giabit`. I addressed this issue via dealing with corner cases override  in `network_performance.sh`.

Reply:
In the near future,  maintainer should be aware of the throughput before release new version if ec2 introduce new  instance types.